language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1451,83 +1455,118 @@ private static String getQualifiedModuleName(Module module) {
+                                                                  PositionedOperationContext ctx,
+                                                                  NodeList<FunctionArgumentNode> arguments,
+                                                                  boolean isLangLibFunction) {
+        Optional<ParameterSymbol> parameterSymbol =
+                resolveParameterSymbol(functionTypeSymbol, ctx, arguments, isLangLibFunction);
+        if (parameterSymbol.isEmpty()) {
             return Optional.empty();
         }
-        return resolveParameterSymbol(functionTypeSymbol, ctx, args.get().arguments());
+        TypeSymbol typeSymbol = parameterSymbol.get().typeDescriptor();
+        if (parameterSymbol.get().paramKind() == ParameterKind.REST && typeSymbol.typeKind() == TypeDescKind.ARRAY) {
+            return Optional.of(((ArrayTypeSymbol) typeSymbol).memberTypeDescriptor());
+        }
+        return Optional.of(typeSymbol);
     }
 
+
     /**
-     * Given the cursor position information, returns the expected ParameterSymbol
-     * information corresponding to the FunctionTypeSymbol instance.
+     * Finds the corresponding function type symbol from lang libs method given a method call expresison node. 
      *
-     * @param functionTypeSymbol Referenced FunctionTypeSymbol
-     * @param ctx                Positioned operation context information.
-     * @param node               Explicit new expression node.
-     * @return {@link Optional<ParameterSymbol>} Expected Parameter Symbol.
+     * @param methodCallExprNode method call expression node.
+     * @param context context
+     * @return {@link Optional<ParameterSymbol>} function type symbol
      */
-    public static Optional<ParameterSymbol> resolveFunctionParameterSymbol(FunctionTypeSymbol functionTypeSymbol,
-                                                                           PositionedOperationContext ctx,
-                                                                           ExplicitNewExpressionNode node) {
-        ParenthesizedArgList args = node.parenthesizedArgList();
-        return resolveParameterSymbol(functionTypeSymbol, ctx, args.arguments());
+    public static Optional<FunctionTypeSymbol> findMethodInLangLibFunctions(MethodCallExpressionNode methodCallExprNode,
+                                                                            PositionedOperationContext context) {
+
+        if (methodCallExprNode.methodName().kind() != SyntaxKind.SIMPLE_NAME_REFERENCE) {
+            return Optional.empty();
+        }
+        SimpleNameReferenceNode typeRefNode = (SimpleNameReferenceNode) methodCallExprNode.methodName();
+        Optional<TypeSymbol> typeSymbol = context.currentSemanticModel()
+                .flatMap(semanticModel -> semanticModel.typeOf(methodCallExprNode.expression()));
+        return typeSymbol.flatMap(value -> value.langLibMethods().stream()
+                .filter(method -> method.getName().isPresent() &&
```

Comment:
In a simpler manner, we can use `method.getName().orElse("").equals(...)`

Reply:
Done